nvaccess/nvda
# JAB: do not treat ActiveDescendentChange event from not focused component as focus gained event (#5989)

### Link to issue number:
closes #5989
### Summary of the issue:
When a PropertyActiveDescendentChange event is received, nvda transform it into a event_gainFocus event.

The issue is that PropertyActiveDescendentChange can be raised by components (e.g. tree view) that don't have the focus. In that case, a PropertyActiveDescendentChange does not imply that the newly selected tree node has received the focus.
### Description of how this pull request fixes the issue:
Treat PropertyActiveDescendentChange event as focus gained event only if event source has focus or it is an ancestor is the focused component.
### Testing performed:
- Start Java application mention in the issue comment.
- Move caret in the opened text editor.
- Ensure that changing tree selection on caret move not treaded as focus gained and
it is not voiced by nvda.
### Known issues with pull request:
None
### Change log entry:
*Bug fixes*
`Do not change focus in Java application when selection is change in not focused tree, table or list. (#5989)`
